src/_P092_DLbus.ino

```diff
@@ -0,0 +1,1180 @@
+
+//#######################################################################################################
+//########################### Plugin 092: DL-bus from Technische Alternative ############################
+//#######################################################################################################
+
+/**************************************************\
+   This plug-in reads and decodes the DL-Bus.
+   The DL-Bus is used in heating control units e.g. sold by Technische Alternative (www.ta.co.at).
+
+   The idea for this plug-in is based on Martin Kropf's project UVR31_RF24 (https://github.com/martinkropf/UVR31_RF24)
+
+   The plug-in is tested and workis fine for the ESR21 device.
+   The plug-in should be also able to decode the information from the
+   UVR31, UVR1611 and UVR 61-3 devices.
+
+   The selected input needs a voltage divider as follows because the DL-Bus runs on 12 volts.
+   DLbus@12V - 8k6 - input@3.3V - 3k3 - ground
+\**************************************************/
+
+#define PLUGIN_092
+#define PLUGIN_ID_092         92
+//#define PLUGIN_092_DEBUG    // additional debug messages in the log
+#define PLUGIN_NAME_092       "Heating - DL-Bus (Technische Alternative)"
+#define PLUGIN_VALUENAME1_092 "Value"
+
+#define P092_DLbus_OptionCount 8
+#define P092_DLbus_ValueCount 1
+#define P092_DLbus_DeviceCount 5
+
+void P092_Pin_changed() ICACHE_RAM_ATTR;
+volatile uint8_t P092_DLB_Pin;
+volatile boolean P092_receiving = false;    // receiving flag
+boolean P092_init = false;
+boolean P092_ReceivedOK = false;
+uint32_t P092_LastReceived = 0;
+uint8_t P092_MaxIdx[P092_DLbus_OptionCount];
+int P092_ValueType[P092_DLbus_ValueCount];
+int P092_ValueIdx[P092_DLbus_ValueCount];
+struct DataStruct
+{
+  uint8_t				Datauint8_ts;
+  uint8_t       Deviceuint8_t0;
+  uint8_t       Deviceuint8_t1;
+  uint8_t       Deviceuint8_ts;
+  uint8_t       DontCareuint8_ts;
+  uint8_t       TimeStampuint8_ts;
+  uint8_t       MaxSensors;
+  uint8_t       MaxExtSensors;
+  uint8_t       Outputuint8_ts;
+  uint8_t       Speeduint8_ts;
+  uint8_t       Analoguint8_ts;
+  uint8_t       Volumeuint8_ts;
+  uint8_t       MaxHeatMeters;
+  uint8_t       CurrentHmuint8_ts;
+  uint8_t       MWhuint8_ts;
+
+  uint16_t      MinPulseWidth;
+  uint16_t      MaxPulseWidth;
+  uint16_t      MinDoublePulseWidth;
+  uint16_t      MaxDoublePulseWidth;
+  uint8_t       IdxSensor;
+  uint8_t       IdxExtSensor;
+  uint8_t       IdxOutput;
+  uint8_t       IdxDrehzahl;
+  uint8_t       IdxAnalog;
+  uint8_t       IdxHmRegister;
+  uint8_t       IdxVolume;
+  uint8_t       IdxHeatMeter1;
+  uint8_t       IdxkWh1;
+  uint8_t       IdxMWh1;
+  uint8_t       IdxHeatMeter2;
+  uint8_t       IdxkWh2;
+  uint8_t       IdxMWh2;
+  uint8_t       IdxHeatMeter3;
+  uint8_t       IdxkWh3;
+  uint8_t       IdxMWh3;
+  uint8_t       IdxCRC;
+} DataSettings;
+
+typedef struct {
+  uint8_t Idx;
+  uint8_t mode;
+  float   value;
+} data_t;
+data_t ReadData;
```

Comment:
Can you wrap these structs in a class or struct specific for this plugin?
Their names are so generic, they are bound to cause a conflict sooner or later with other code.
You have them global defined, so they are accessible by all other code.

Reply:
Done!